### PR TITLE
Fix NPE in cpu profile

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Sometimes we can get a traceEvent with a stack frame id that we don't actually have a stackFrame for.

In my observation, this is usually the trace event for what should be the first stack frame in the response. I'm talking to @bkonyi about this, but this will protect against the NPE for now. I haven't seen this happen until recently so maybe it is a recent regression with the path _getCpuProfileTimeline triggers.

### CODE REVIEW

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart [hunk 1]
```diff
@@ -122,7 +122,7 @@ class CpuProfileData {
   void _setExclusiveSampleCounts() {
     for (Map<String, dynamic> traceEvent in stackTraceEvents) {
       final leafId = traceEvent[stackFrameId];
-      stackFrames[leafId].exclusiveSampleCount++;
+      stackFrames[leafId]?.exclusiveSampleCount++;
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: why do we have a case where the leafId is missing?
Can we add a test case that fails without this check and succeeds with the check?
I'm worried this change means we are quietly dropping frames if we don't find the expected ID.
Is it possible we are sometimes getting data out of order in a way that causes this issue?

Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: I asked ben about it, but I haven't heard back. It's not an ordering issue in this case - the stack frame just isn't in the response when it should be.

I agree we don't want to be silently dropping frames, but this check is to protect ourselves against bad data so it is necessary for now at least. Hopefully Ben will be able to help us identify the root cause and we can revert this check.

Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: I added a test. I was having a hard time repro-ing so I added an assert with instructions to export the timeline and send it to me if anyone else hits this while working on DevTools.

**Comment by kenzieschmoll** [comment]
Landing this. https://github.com/flutter/devtools/pull/672 fixes the inspector golden failure.

**Pull request merged**